content/en/docs/concepts/overview/object-management-kubectl/kustomization.md

```diff
@@ -0,0 +1,752 @@
+---
+title: Declarative Management of Kubernetes Objects Using Kustomization
+content_template: templates/concept
+weight: 40
+---
+
+{{% capture overview %}}
+[Kustomize](https://github.com/kubernetes-sigs/kustomize) is a standalone tool
+to customize Kubernetes objects
+through a [kusotmization file](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/kustomization.yaml). 
+Since 1.14, Kubectl also
+supports the management of Kubernetes objects using a kustomization file.
+When dealing with Resources or objects from kustomization, the flag
+`--kusotmize` or `-k` need to be used in Kubectl commands.
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Overview of Kustomize
+Kustomize is a tool for customizing Kubernetes configuration. 
+It offers simple tooling to manage configuration files for your applications with features from aspects as
+* generating resources from other sources
+* setting cross-cutting fields for resources
+* composing and customizing collections of resources
+
+### Generating Resources
+ConfigMap and Secrets hold config or sensitive data that are used by other Kubernetes objects, such as Pods. The source
+of truth of ConfigMap or Secrets are usually from somewhere else, such as a `.properties` file or a ssh key file.
+Kustomize has `secretGenerator` and `configMapGenerator`, which generate Secrets and ConfigMaps from files or literals.
+
+
+#### configMapGenerator
+To generate a ConfigMap from a file, add an entry to `files` list in `configMapGenerator`. Here is an example of generating a ConfigMap with a data item from a file content.
+```shell
+# Create a application.properties file
+cat <<EOF >application.properties
+FOO=Bar
+EOF
+
+cat <<EOF >./kustomization.yaml
+configMapGenerator:
+- name: example-configmap-1
+  files:
+  - application.properties
+EOF
+```
+The generated ConfigMap can be checked by following command
+```shell
+kubectl kustomize ./
+```
+The generated ConfigMap is
+```yaml
+apiVersion: v1
+data:
+  application.properties: |
+    FOO=Bar
+kind: ConfigMap
+metadata:
+  name: example-configmap-1-8mbdf7882g
+```
+
+ConfigMap can also be generated from literal key-value pairs. To generate a ConfigMap from a literal key-value pair, add an entry to `literal` list in configMapGenerator. Here is an example of generating a ConfigMap with a data item from a key-value pair.
+```shell
+cat <<EOF >./kustomization.yaml
+configMapGenerator:
+- name: example-configmap-2
+  literals:
+  - FOO=Bar
+EOF
+```
+The generated ConfigMap can be checked by following command
+```shell
+kubectl kustomize ./
+```
+The generated ConfigMap is
+```yaml
+apiVersion: v1
+data:
+  FOO: Bar
+kind: ConfigMap
+metadata:
+  name: example-configmap-2-g2hdhfc6tk
+```
+
+#### secretGenerator
+Secrets can also be generated from files or literal key-value paris. To generate a Secret from a file, add an entry to `files` list in `secretGenerator`. Here is an example of generating a Secret with a data item from a file.
+```shell
```

Comment:
spelling: key-value pairs.

Reply:
nit: line 108,
The generated Secret is as following:
s/following/follows

Reply:
updated